src/CodeGen_Vulkan_Dev.cpp

```diff
@@ -2120,53 +2129,65 @@ void CodeGen_Vulkan_Dev::SPIRV_Emitter::visit(const Shuffle *op) {
 
     } else {
 
-        // vector shuffle ... not interleaving
-        int op_lanes = op->type.lanes();
-        int num_vectors = (int)op->vectors.size();
-
-        std::cout << "    vector shuffle x" << num_vectors << " : ";
-        for (int idx : op->indices) {
-            std::cout << idx << " ";
+        // Vector shuffle with arbitrary number of lanes per arg
+        //
+        // We need to construct the mapping between shuffled-index,
+        // and source-vector-index and source-element-index-within-the-vector.
+        //
+        // To start, we'll figure out what the first shuffle-index is per
+        // source-vector. Also let's compute the total number of
+        // source-elements the to be able to assert that all of the
+        // shuffle-indices are within range.
+        //
+        std::vector<int> vector_first_index;
+        int max_index = 0;
+        for (const Expr &v : op->vectors) {
+            vector_first_index.push_back(max_index);
+            max_index += v.type().lanes();
+        }
+        for (int i : op->indices) {
+            internal_assert(i >= 0 && i < max_index);
```

Comment:
Ah, you have this sanity check. Nice. So, I suggest promoting it to be executed in all scenarios (also when it's not an arbitrary shuffle).